lib/exprengine.cpp

```diff
@@ -2646,19 +2647,30 @@ static std::string execute(const Token *start, const Token *end, Data &data)
                         data.assignStructMember(tok2, &*structVal, memberName, memberValue);
                         continue;
                     }
+
+                    // Assign a pointer value
                     if (lhs->isUnaryOp("*") && lhs->astOperand1()->varId()) {
                         const Token *varToken = tok2->astOperand1()->astOperand1();
                         ExprEngine::ValuePtr val = data.getValue(varToken->varId(), varToken->valueType(), varToken);
-                        if (val && val->type == ExprEngine::ValueType::ArrayValue) {
-                            // Try to assign "any" value
-                            auto arrayValue = std::dynamic_pointer_cast<ExprEngine::ArrayValue>(val);
-                            arrayValue->assign(std::make_shared<ExprEngine::IntRange>("0", 0, 0), std::make_shared<ExprEngine::BailoutValue>());
+                        if (!val || val->type != ExprEngine::ValueType::ArrayValue)
+                            throw ExprEngineException(tok2, "Unhandled assignment in loop");
```

Comment:
For information..

In general I try to avoid such exception and use BailoutValue instead.

When this exception is thrown.. all the dataflow analysis is stopped and there will be all possible false positives. With a `BailoutValue` we should get the same false positives but only for a certain variable..

Reply:
I see. But I can't assign `BailoutValue` in this case, because I don't have a value to change.

Reply:
ok yes I agree.